### ISSUE TITLE
context or contextFilter expressions in changesets are no longer filtered from liquibase-maven-plugin version 4.22.0 onwards

### ISSUE THREAD

**Issue by ivanfernandezetu** [liquibase/liquibase#5197]
### Search first

- [X] I searched and no similar issues were found

### Description

Hi,

We use Liquibase Contexts in our SQL changesets to specify changes that should only apply to test environments:

```
--liquibase formatted sql

--changeset ibanFR:1  context:prod
create table person (
    id int primary key auto_increment not null,
    name varchar(50) not null,
    address1 varchar(50),
    address2 varchar(50),
    city varchar(30)
)
--rollback DROP TABLE person;

--changeset ibanFR:2 context:test
--comment: insert test data
insert into person (name) values ("ivan")
```

Quite recently we have upgraded the liquibase-maven-plugin version to the latest version available 4.24.0 and we have realised that changes containing context expressions tried to be applied when running liquibase status command:

`mvn liquibase:status --Dliquibase.contexts=prod`

```
2 changesets have not been applied to SA@jdbc:h2:tcp://localhost:9092/~/test
     example-changelog.sql::1::ibanFR
     example-changelog.sql::2::ibanFR
```

After testing previous versions we noticed that this functionality stopped working for version 4.22.0 of the liquibase-maven-plugin

### Steps To Reproduce

1. Create a sample changelog containing the following changesets:

```
--liquibase formatted sql

--changeset ibanFR:1  context:prod
create table person (
    id int primary key auto_increment not null,
    name varchar(50) not null,
    address1 varchar(50),
    address2 varchar(50),
    city varchar(30)
)
--rollback DROP TABLE person;

--changeset ibanFR:2 context:test
--comment: insert test data
insert into person (name) values ("ivan")
```

2. Run a liquibase status command activating prod context:

`mvn liquibase:status --Dliquibase.contexts=prod` 

### Expected/Desired Behavior

Only changesets containing prod context should be marked to be applied to the target database:

```
1 changesets have not been applied to SA@jdbc:h2:tcp://localhost:9092/~/test
     example-changelog.sql::1::ibanFR
```

### Liquibase Version

4.24.0

### Database Vendor & Version

H2 2.2.224, MySQL 8.1.0

### Liquibase Integration

maven

### OS and/or Infrastructure Type/Provider

MacOS Sonoma 14.0

### Additional Context

A sample project has been created to isolate this behavior: https://github.com/ibanFR/liquibase-maven-demo.git

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by tati-qalified** [comment]
Hi @ivanfernandezetu, thank you for reporting this issue.
I have been able to verify it, and it's still replicable with Liquibase v4.25.0.

I'll be forwarding this to the development team so they can evaluate how to move forward.
Thank you,
Tatiana

**Comment by ComputerDaddyGuy** [comment]
Hello,

I encounter the same issue, as liquibase maven plugin version `4.24.0` is targeted by Spring Boot `3.2.x` ([source](https://github.com/spring-projects/spring-boot/blob/3.2.x/spring-boot-project/spring-boot-dependencies/build.gradle#L831)), so I have to keep maven property `<liquibase.version>4.20.0</liquibase.version>` to keep it working with contexts.

Any idea how we could fix this? I would be happy to help, if possible :)

Thanks!

**Comment by filipelautert** [comment]
Hello @ivanfernandezetu @ComputerDaddyGuy @tati-qalified 
Could you try to use environment variable LIQUIBASE_CONTEXTS instead of the -D parameter and confirm if it works? I'm suspecting that it may be something else with -D parameters. Thanks!

**Comment by tati-qalified** [comment]
Hi @filipelautert, I'm getting some strange behaviour with the environment variables. For reference, I'm using Windows.
First I set my environment variable as `LIQUIBASE_CONTEXTS=prod`. 
Then in my pom.xml file, in the `liquibase-maven-plugin` element, I added this:
```
<configuration>
	<propertyFileWillOverride>true</propertyFileWillOverride>
	<propertyFile>target/classes/liquibase.properties</propertyFile>
	<contexts>${env.LIQUIBASE_CONTEXTS}</contexts>
</configuration>
```

When I run `mvn liquibase:status`, the context is not taken into consideration, and the output shows this:
```
[WARNING]

Liquibase detected the following invalid LIQUIBASE_* environment variables:

- LIQUIBASE_CONTEXTS

Find the list of valid environment variables at https://docs.liquibase.com/environment-variables
```

When checking the provided [list of environment variables](https://docs.liquibase.com/environment-variables), it appears that the correct variable name for  Liquibase v4.23.0+ is `LIQUIBASE_COMMAND_CONTEXT_FILTER`.

I tried using this new name - I updated my environment variable and I tried to use `<contexts>${env.LIQUIBASE_COMMAND_CONTEXT_FILTER}</contexts>`, but it's not recognised as an option.
![image](https://github.com/liquibase/liquibase/assets/135264316/32a8519e-1f8c-41bf-abc0-793b498be604)

The context is still not recognised and the same error message is shown - still saying that the invalid environment variable is `LIQUIBASE_CONTEXT`.

Any ideas on how to move forward?

**Comment by filipelautert** [comment]
Hi @tati-qalified I took a better look at this and it's a hell of renamed/duplicated properties.

From update help command, the property should be `contextFilter` .  I can see that contexts is listed as a legacy property for liquibase command line too, so somewhere down the road it was renamed (probably at 4.24.0?).
But there is also maven property `contexts` - that's another thing, and it should still work. 

So any of those options should work:

1. Using -Dliquibase.contextFilter=lalala
2. Using tag `<context>${env.A_RANDOM_VAR_THAT_DOES_NOT_START_WITH_LIQUIBASE_UNDERLINE</context>`
3. (Not sure about this one) Using `-Dcontexts=lalala `

**Comment by natros** [comment]
Is there any news on this issue? I'm testing with 4.30.0 and the `contextFilter` is not working as expected.

After a little debugging session it seems there is a missing property in `liquibase.Liquibase#reportStatus(boolean, liquibase.Contexts, liquibase.LabelExpression, java.io.Writer)`. There is no `statusCommand.addArgumentValue(DatabaseChangelogCommandStep.CONTEXTS_ARG, (contexts != null? contexts.toString() : null))` to set the context. Other commands have it defined.

**Comment by filipelautert** [comment]
> Is there any news on this issue? I'm testing with 4.30.0 and the `contextFilter` is not working as expected.
> 
> After a little debugging session it seems there is a missing property in `liquibase.Liquibase#reportStatus(boolean, liquibase.Contexts, liquibase.LabelExpression, java.io.Writer)`. There is no `statusCommand.addArgumentValue(DatabaseChangelogCommandStep.CONTEXTS_ARG, (contexts != null? contexts.toString() : null))` to set the context. Other commands have it defined.

Makes sense! but instead of using this parameter I fixed CHANGELOG_PARAMETERS  - it's a bit bizarre that we have 2 ways of passing the same parameter, but now both works.
